### ISSUE TITLE
Ruff complains about a too long line which black does not format

### ISSUE THREAD

**Issue by nbro10** [astral-sh/ruff#5385]
Ruff complains about this (in file `issue.py`), although black doesn't change this code
```

class A:
    def my_func(self, constants):
        headers = {
            constants.HEADERS.CONTENT_TYPE.value: constants.CONTENT_TYPE.URL_ENCODED.value
        }

```

The line is ofc `            constants.HEADERS.CONTENT_TYPE.value: constants.CONTENT_TYPE.URL_ENCODED.value`, which has 90 chars

Ruff version: 0.0.275
Invoked command: `ruff check issue.py`
No configuration added to pyproject.toml for either ruff or black (so both should assume a max line-length of 88)

**Comment by charliermarsh** [comment]
Ah yeah, this is by-design and consistent with Flake8 and other linters. In general, the line-length check is just looking at the length of the physical line, and not whether a formatter has the ability to truncate it. If you want to rely on Black to enforce line length, you might consider disabling E501 altogether.

**Comment by charliermarsh** [comment]
It's compatible with Black in that it won't make any changes that conflict with Black, and there aren't any rules that will be triggered by way of running Black on your codebase. (This isn't true of Flake8 or Pycodestyle, which have rules that disagree with Black's formatting.)

**Comment by charliermarsh** [comment]
Can I ask a slightly different question here: what is your ideal behavior?

**Comment by dimaqq** [comment]
> Ah yeah, this is by-design and consistent with Flake8 and other linters. In general, the line-length check is just looking at the length of the physical line, and not whether a formatter has the ability to truncate it. If you want to rely on Black to enforce line length, you might consider disabling E501 altogether.

Sorry to chime in, as a non-user of `ruff`, rather someone who is trying it out, it seems like the tool is taking on too much... I'd be very happy to leave formatting to `black` entirely and have this tool focus on all the other fluff.

**Comment by dimaqq** [comment]
Here's another example:

```py
async def frob(foo):                                                                                                                                    
    match foo.bar:                                                                        
        case 42:    
            try:            
                httpx.do.something()                                                    
            except httpx.HTTPStatusError as e:                                          
                logging.exception(                                                      
                    f"Error response {e.response.status_code!r} while requesting {e.request.url!r}"
                )                      
```

This kind of line could only be split using C-style string literal concatenation:
```
                logging.exception(                                                      
                    "Error response "
                    f"{e.response.status_code!r} while requesting {e.request.url!r}"
                )  
```

This has been discussed at `black` a while back and ultimately a sensible decision was taken not to split user literals.

Which leaves long literals a valid exception to the line length rule.

I think long type annotations and complex inline dicts are the same.

Please consider re-opening this ticket.

**Comment by charliermarsh** [comment]
We enable only a very few stylistic rules by default. You’re welcome to turn off the line-length rule if you find it unhelpful! Though note that Black alone does not solve line lengths — you can still have overlong comments, strings, lines that Black doesn’t know how to split, and so on. I think it’s very reasonable to have a linter that checks for overlong lines (_not_ supporting a line length check would strike me as unusual, Flake8 has one too, etc.), but again, you’re welcome to turn it off.

**Comment by charliermarsh** [comment]
What I don’t quite follow is: if you want Black to be the determiner of what’s a valid line length, why enable E501 at all?

**Comment by charliermarsh** [comment]
(To be clear, we don’t attempt to autofix or modify line lengths at all.)

**Comment by dimaqq** [comment]
I'd like to have good tools work together out of the box.

In fact, I've `flake8` from a few repos merely because the tool required custom config to be compatible with `black`.

It is true, too, that another, more pedantic developer brought `flake8` with the correct set of disabled E's and W's, so not all is lost.

My humble 2c: a young, lesser known project must be compatible from get go to gain traction.

**Comment by zanieb** [comment]
@nbro10 I think this is clearly documented at https://beta.ruff.rs/docs/faq/#is-ruff-compatible-with-black — is there somewhere else you think this information should be highlighted?

Black cannot safely automatically split some long lines. In these cases, it is valuable to many users to have a linter highlight the line length violation so it can be split manually.

**Comment by eli-schwartz** [comment]
> If you're saying that Ruff will not report anything after running black, provided that we first fix all issues reported by ruff before running black itself, this may not be true, because our fix may be undone by black?

Are you saying that running `black` caused two shorter lines to be merged into one longer line?
